### PR TITLE
Api doc update

### PR DESCRIPTION
**Pull request by ChandraDevs1**
What’s Added:

New Configuration Classes section in API.rst.

Documentation for dataclasses: BaseConfig, LegendConfig, BasePlotConfig, LineConfig, VLineConfig, ScatterConfig, ChromatogramConfig, MobilogramConfig, SpectrumConfig, PeakMapConfig

resolves https://github.com/OpenMS/pyopenms_viz/issues/52

@jcharkow please review the PR and let me know for any changes 

## Summary by CodeRabbit

- **Documentation**
  - Improved clarity and consistency of the API documentation with corrected typos and updated section formatting.
  - Introduced a dedicated section outlining various configuration classes.

- **New Features**
  - Expanded visualization capabilities with enhanced plotting support for chromatograms, spectra, mobilograms, and peakmaps.

- **Tests**
  - Added comprehensive tests to ensure robust handling of edge cases for the new plotting features.
  - Removed outdated tests for mobilogram, spectrum, and peakmap plotting functionality.

### CODE REVIEW

**Comment by jcharkow** [comment]
Is this replacing the other PR? For next time it is best practice to continue with the old PR so that the conversation history can be saved. Also if you need to create another PR it is best to link the old PR. 

It seems that the documentation build is currently failing but it does not seem to be related to your code. 

Will leave a full review soon

**Comment by ChandraDevs1** [comment]
> Is this replacing the other PR? If it is can you please close the old PR? For next time it is best practice to continue with the old PR so that the conversation history can be saved. Also if you need to create another PR it is best to link the old PR.
> 
> It seems that the documentation build is currently failing but it does not seem to be related to your code.
> 
> Will leave a full review soon
sure and thanks for guiding me

File: docs/API.rst [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
+   BaseConfig
+   LegendConfig
+   BasePlotConfig
+   LineConfig
+   VLineConfig
+   ScatterConfig
+   ChromatogramConfig
+   MobilogramConfig
+   SpectrumConfig
+   PeakMapConfig
```
Comment (jcharkow) [inline docs/API.rst:25]: I would put this at the bottom of the API documentation, does not need to be this prevalent.

File: pyopenms_viz/_core.py [hunk 2]
```diff
@@ -1500,3 +1499,187 @@ def _get_plot_backend(backend: str | None = None):
...
+    y : str
+        The column name for the y-axis data.
+    backend : str, optional
+        The backend to use for plotting. If not provided, it will try to use
+        matplotlib, bokeh, or plotly in that order.
+    **kwargs
+        Additional keyword arguments passed to the plotting function. Supported arguments depend on the backend:
+        - For `ms_matplotlib`: See matplotlib documentation.
+        - For `ms_bokeh`: See bokeh documentation.
+        - For `ms_plotly`: See plotly documentation.
```
Comment (jcharkow) [inline pyopenms_viz/_core.py:1523]: This is not true however kwargs, are specific parameters for the pyopenms_viz function. You can link to the parameters section of the documentation though!

File: pyopenms_viz/_core.py [hunk 3]
```diff
@@ -1500,3 +1499,187 @@ def _get_plot_backend(backend: str | None = None):
...
+    ----------
+    data : pandas.DataFrame
+        The data to plot.
+    x : str
+        The column name for the x-axis data.
+    y : str
+        The column name for the y-axis data.
+    backend : str, optional
+        The backend to use for plotting. If not provided, it will try to use
+        matplotlib, bokeh, or plotly in that order.
```
Comment (jcharkow) [inline pyopenms_viz/_core.py:1518]: This functionality described does not seem to be implemented. Can you show me where this is implemented? If it is not  implement this or make backend required argument. e.g. backed: Literal["ms_plotly", "ms_bokeh", "ms_matplotlib"]

File: pyopenms_viz/_core.py [hunk 4]
```diff
@@ -1500,3 +1499,187 @@ def _get_plot_backend(backend: str | None = None):
...
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import pyopenms_viz as oms_viz
+    >>> data = pd.DataFrame({'rt': [1, 2, 3], 'intensity': [10, 20, 30]})
+    >>> oms_viz.plot_chromatogram(data, x='rt', y='intensity', backend='ms_plotly')
+    """
+    if x is None or y is None:
+        raise ValueError("Both `x` and `y` must be provided.")
```
Comment (jcharkow) [inline pyopenms_viz/_core.py:1543]: This code is a bit redundant because both x and y are required so now so it is unlikely they will be none. Can remove this.

**Review by jcharkow — CHANGES_REQUESTED** [review]
Good progress, thank you for addressing the requested changes. Please see comments above. 

Also it seems that the ci tests are not triggering on github. I will try to fix that and we can then check if the new tests you added are working as intended.

**Comment by jcharkow** [comment]
Please also merge the main branch into this branch as this should trigger tests to run.

File: pyopenms_viz/_core.py (hunk 4 above)
Comment (ChandraDevs1) [reply pyopenms_viz/_core.py]: done

File: pyopenms_viz/_core.py (hunk 3 above)
Comment (ChandraDevs1) [reply pyopenms_viz/_core.py]: done

File: pyopenms_viz/_core.py (hunk 2 above)
Comment (ChandraDevs1) [reply pyopenms_viz/_core.py]: @jcharkow  ✅

**Comment by ChandraDevs1** [comment]
New One PR AND Updated One  :https://github.com/OpenMS/pyopenms_viz/pull/77
